### PR TITLE
💪 Typesafe pick, omit and pluck

### PR DESCRIPTION
**Pull request by nicolasgramlich**

### CODE REVIEW

File: src/utils/omit.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import * as Lodash from 'lodash';
+
+export type Diff<T extends string, U extends string> = ({[P in T]: P } & {[P in U]: never } & { [x: string]: never })[T];
```
Comment (SylvainEstevez) [inline src/utils/omit.ts]: TDiff?

**Review by SylvainEstevez — APPROVED** [review]
LGTM beside naming convention

File: src/utils/omit.ts (hunk 1 above)
Comment (nicolasgramlich) [reply src/utils/omit.ts]: Don't you think it's nicer like this, because it blends in amongst `Pick<>` and co?

Comment (SylvainEstevez) [reply src/utils/omit.ts]: Makes sense indeed, could expose those as types BTW?

Comment (nicolasgramlich) [reply src/utils/omit.ts]: You mean move them to the `types` directory?

**Comment by nicolasgramlich** [comment]
Travis build is broken due to: https://github.com/npm/npm/issues/17858

File: src/utils/omit.ts (hunk 1 above)
Comment (SylvainEstevez) [reply src/utils/omit.ts]: Yes

Comment (nicolasgramlich) [reply src/utils/omit.ts]: Done

**Comment by nicolasgramlich** [comment]
Build: 🍏  after removing node_modules from cache. This repo is so small we'll be fine without caching

**Pull request merged**